### PR TITLE
cleanup federated resource quota

### PR DESCRIPTION
**Pull request by calvin0327**
Signed-off-by: calvin0327 <wen.chen@daocloud.io>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
tidy federated resource quota code.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
please correct me if not

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by calvin0327** [comment]
https://github.com/karmada-io/karmada/runs/7169278909?check_suite_focus=true#step:5:81

The e2e still seems to have a bit of a problem. the member cluster is not ready?

File: pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go [hunk 1]
```diff
@@ -139,10 +135,7 @@ func (c *SyncController) cleanUpWorks(namespace, name string) error {
 		}
 	}
 
-	if len(errs) > 0 {
-		return errors.NewAggregate(errs)
-	}
-	return nil
```
Comment (XiShanYongYe-Chang) [inline pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go:145]: There are other files coded like this, updated together?

Comment (calvin0327) [reply pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go:145]: Done.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks for your cleanup~

/lgtm

File: pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go [hunk 2]
```diff
@@ -190,16 +183,13 @@ func (c *SyncController) buildWorks(quota *policyv1alpha1.FederatedResourceQuota
...
+	return errors.NewAggregate(errs)
 }
 
 func extractClusterHardResourceList(spec policyv1alpha1.FederatedResourceQuotaSpec, cluster string) corev1.ResourceList {
-	for index := range spec.StaticAssignments {
-		if spec.StaticAssignments[index].ClusterName == cluster {
-			return spec.StaticAssignments[index].Hard
+	for _, staticAssignment := range spec.StaticAssignments {
+		if staticAssignment.ClusterName == cluster {
+			return staticAssignment.Hard
```
Comment (RainbowMango) [inline pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go:192]: Why?

**Review by RainbowMango — COMMENTED** [review]
can you explain `why we need it`?
Sorry, I don't see much benefit.

File: pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go (hunk 2 above)
Comment (RainbowMango) [reply pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go:192]: I don't think so. It's a kind of coding style. I don't think we need to work on that.

File: pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go [hunk 3]
```diff
@@ -120,12 +119,9 @@ func (c *SyncController) SetupWithManager(mgr controllerruntime.Manager) error {
...
 	workList := &workv1alpha1.WorkList{}
-	if err := c.List(context.TODO(), workList, &client.ListOptions{
-		LabelSelector: labels.SelectorFromSet(
-			labels.Set{
-				util.FederatedResourceQuotaNamespaceLabel: namespace,
-				util.FederatedResourceQuotaNameLabel:      name,
-			}),
+	if err := c.List(context.TODO(), workList, client.MatchingLabels{
+		util.FederatedResourceQuotaNamespaceLabel: namespace,
+		util.FederatedResourceQuotaNameLabel:      name,
```
Comment (RainbowMango) [inline pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go:124]: This looks better.

File: pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go (hunk 2 above)
Comment (calvin0327) [reply pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go:192]: ok. thanks

**Comment by calvin0327** [comment]
@RainbowMango done.

File: pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go [hunk 4]
```diff
@@ -139,10 +135,7 @@ func (c *SyncController) cleanUpWorks(namespace, name string) error {
 		}
 	}
 
-	if len(errs) > 0 {
-		return errors.NewAggregate(errs)
-	}
-	return nil
+	return errors.NewAggregate(errs)
```
Comment (RainbowMango) [inline pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go:138]: If you want to unify the style, then please update all cases from this repo.

**Comment by XiShanYongYe-Chang** [comment]
The CI run failed for setting up Karmada env, this has nothing to do with this pr, we need to retest the CI.

**Comment by calvin0327** [comment]
image

@XiShanYongYe-Chang Our e2e env is seem not very stable. Is there not enough server resources?

**Comment by calvin0327** [comment]
@RainbowMango ok, this has all been amend.

**Comment by XiShanYongYe-Chang** [comment]
> Our e2e env is seem not very stable. Is there not enough server resources?

I'm not sure. Since we supported v1.24 k8s in our ci test, this similar error often happens, the `karmada-apiserver` has restarted multiple times.

Error log info:

```log
2022-07-07T05:22:46.116588292Z stderr F Flag --insecure-port has been deprecated, This flag has no effect now and will be removed in v1.24.
2022-07-07T05:22:46.160071829Z stderr F I0707 05:22:46.151332       1 server.go:565] external host was not specified, using 172.18.0.4
2022-07-07T05:22:46.190172825Z stderr F I0707 05:22:46.173055       1 server.go:172] Version: v1.23.8
2022-07-07T05:23:05.055655899Z stderr F I0707 05:23:05.055056       1 shared_informer.go:240] Waiting for caches to sync for node_authorizer
2022-07-07T05:23:05.155979012Z stderr F I0707 05:23:05.152330       1 plugins.go:158] Loaded 10 mutating admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,NodeRestriction,TaintNodesByCondition,Priority,DefaultTolerationSeconds,DefaultStorageClass,RuntimeClass,DefaultIngressClass,MutatingAdmissionWebhook.
2022-07-07T05:23:05.328014449Z stderr F I0707 05:23:05.152870       1 plugins.go:161] Loaded 10 validating admission controller(s) successfully in the following order: LimitRanger,PodSecurity,Priority,PersistentVolumeClaimResize,RuntimeClass,CertificateApproval,CertificateSigning,CertificateSubjectRestriction,ValidatingAdmissionWebhook,ResourceQuota.
2022-07-07T05:23:05.372089386Z stderr F I0707 05:23:05.371927       1 plugins.go:158] Loaded 10 mutating admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,NodeRestriction,TaintNodesByCondition,Priority,DefaultTolerationSeconds,DefaultStorageClass,RuntimeClass,DefaultIngressClass,MutatingAdmissionWebhook.
2022-07-07T05:23:05.372122787Z stderr F I0707 05:23:05.371967       1 plugins.go:161] Loaded 10 validating admission controller(s) successfully in the following order: LimitRanger,PodSecurity,Priority,PersistentVolumeClaimResize,RuntimeClass,CertificateApproval,CertificateSigning,CertificateSubjectRestriction,ValidatingAdmissionWebhook,ResourceQuota.
2022-07-07T05:23:25.512818348Z stderr F E0707 05:23:25.500988       1 run.go:74] "command failed" err="context deadline exceeded"
```

**Review by RainbowMango — COMMENTED** [review]
/lgtm

**Pull request merged**